src/js/github-api.js

```diff
@@ -20,3 +22,63 @@ export async function isProviderTokenValid(token) {
     return false;
   }
 }
+
+/**
+ * Fetch pull request data from specified parameters.
+ * @param { string } token The SSO token generated
+ * @param { string } owner The owner of the repo
+ * @param { string } repo The repo that the user wants to pull from
+ * @returns { string } returns the pull requests data in json format
+ */
+export async function getPullRequests(token, owner, repo) {
+  const url = `https://api.github.com/repos/${owner}/${repo}/pulls`;
+
+  try {
+    const response = await fetch(url, {
+      headers: {
+        'Authorization': `${token}`,
+        'X-GitHub-Api-Version': '2022-11-28'
+      }
+    });
+    if (!response.ok) {
+      throw new Error('Failed to fetch pull requests.');
+    }
+
+    const data = await response.json();
+    return data;
+
+  }
+  catch (error) {
+    console.log(error);
```

Comment:
Try to avoid leaving ```console``` statements in when merging to release.

Reply:
Replaced with console.error similar to David's function.